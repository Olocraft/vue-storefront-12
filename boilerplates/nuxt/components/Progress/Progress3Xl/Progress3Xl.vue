<template>
  <div
    role="progressbar"
    :aria-valuenow="value"
    aria-valuemin="0"
    aria-valuemax="100"
    aria-label="Loading"
  >
    <!-- eslint-disable max-len -->
    <svg
      width="192"
      height="192"
      stroke-width="8"
      viewPort="0 0 24 24"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <circle
          r="88"
          cx="96"
          cy="96"
          stroke="#e6e6e6"
          fill="none"
          stroke-dasharray="552.64, 552.64"
          class="origin-center rotate-[-90deg]"
        />
        <circle
          r="88"
          cx="96"
          cy="96"
          stroke="#16A34A"
          fill="none"
          :stroke-dasharray="`${552.64 * (value / 100)}, 552.64`"
          class="transition-all ease-in-out duration-[600ms] origin-center rotate-[-90deg]"
        />
        <text
          text-anchor="middle"
          alignment-baseline="central"
          class="text-2xl font-medium origin-center stroke-0 fill-gray-900 font-body stroke-black"
          x="50%"
          y="50%"
        >{{ value }}%</text>
      </g>
    </svg>
    <!-- eslint-enable max-len -->
  </div>
</template>

<script>

export default {
  name: 'Progress3Xl',
  props: {
    value: {
      type: Number,
      default: 0,
    },
  },
};
</script>
