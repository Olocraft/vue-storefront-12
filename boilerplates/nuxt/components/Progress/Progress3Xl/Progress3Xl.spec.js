import { shallowMount } from '@vue/test-utils';
import Progress3Xl from './Progress3Xl.vue';

let wrapper;

describe('Progress3Xl.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(Progress3Xl);
  });
  it('render component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
