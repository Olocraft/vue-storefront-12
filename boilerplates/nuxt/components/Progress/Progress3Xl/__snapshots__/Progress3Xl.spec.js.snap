// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress3Xl.vue render component correctly 1`] = `
<div role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" aria-label="Loading"><svg width="192" height="192" stroke-width="8" viewPort="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg">
    <g>
      <circle r="88" cx="96" cy="96" stroke="#e6e6e6" fill="none" stroke-dasharray="552.64, 552.64" class="origin-center rotate-[-90deg]"></circle>
      <circle r="88" cx="96" cy="96" stroke="#16A34A" fill="none" stroke-dasharray="0, 552.64" class="transition-all ease-in-out duration-[600ms] origin-center rotate-[-90deg]"></circle> <text text-anchor="middle" alignment-baseline="central" x="50%" y="50%" class="text-2xl font-medium origin-center stroke-0 fill-gray-900 font-body stroke-black">0%</text>
    </g>
  </svg></div>
`;
